src/components/demos/write/Demo.js

```diff
@@ -21,138 +24,115 @@ const Wrapper = styled.div`
-  const { theme } = demoState
+  const { update, writeDemo } = state
+  const { theme, showHelp, usernamePasswordPairs, output, showConsole } =
+    writeDemo
+  const demoState = { ...props, ...writeDemo }
   const isMorello = theme.name === 'Morello'
-
-  const [animateLoginFailed, setAnimateLoginFailed] = useState(false)
-  const [showHelp, setShowHelp] = useState(false)
-  const [awaitingApi, setAwaitingApi] = useState(false)
-  const [apiOutput, setApiOutput] = useState('')
-  const [usernamePasswordPairs, setUsernamePasswordPairs] = useState([])
-
-  const resetStates = () => {
-    setShowHelp(false)
-    setAwaitingApi(false)
-    setApiOutput('')
-    setUsernamePasswordPairs([])
-  }
+  const bin = `${binaryName}-${theme.arch}`
+  const [animateLoginFailed, setAnimateLoginFailed] = React.useState(false)
 
   const switchToMorello = (e) => {
     e.preventDefault()
-    resetStates()
     update({
       writeDemo: {
-        ...demoState,
+        ...writeDemo,
+        showHelp: false,
+        output: undefined,
+        showConsole: false,
+        usernamePasswordPairs: [],
         theme: Themes('Morello'),
       },
     })
   }
 
-  useEffect(() => {
-    update(initState)
-  }, [update])
-
   useEffect(() => {
     const attemptLogin = async () => {
-      setAwaitingApi(true)
-      const output = await execute(
-        `${binaryName}-${theme.arch}`,
-        usernamePasswordPairs
-      )
-      setApiOutput(output)
-      setAwaitingApi(false)
+      const output = await execute(bin, usernamePasswordPairs)
+      update({
+        isFetching: false,
+        writeDemo: {
+          ...writeDemo,
+          usernamePasswordPairs: [],
+          output,
+        },
+      })
     }
 
     if (usernamePasswordPairs.length > 0) {
       attemptLogin()
     }
-  }, [usernamePasswordPairs, execute, binaryName, theme])
+    // TODO address demoState dependency, gets trapped in re-rendering state
+  }, [bin, execute, writeDemo, update, usernamePasswordPairs])
+
+  useEffect(() => {
+    update(initState)
+  }, [update])
 
   useEffect(() => {
-    if (isMorello && (failedLogin(apiOutput) || loginError(apiOutput))) {
+    if (isMorello && (failedLogin(output) || loginError(output))) {
       setAnimateLoginFailed(true)
       setTimeout(() => setAnimateLoginFailed(false), 1000)
     }
-  }, [apiOutput, isMorello, setAnimateLoginFailed])
+  }, [output, isMorello, setAnimateLoginFailed])
 
   return (
     <>
       <Header {...props} showClose={true} />
       <Wrapper {...theme.wrapper}>
-        {successfulLogin(apiOutput) ? (
-          <SecretDesktop icons={props.secretDesktop} />
+        {successfulLogin(output) ? (
+          <SecretDesktop font={theme.font} icons={props.secretDesktop} />
         ) : (
-          <>
-            <Help
-              theme={theme}
-              content={helpContent}
-              showContentState={showHelp}
-              setShowContentState={setShowHelp}
-            />
```

Comment:
what has happened to `Help`? Looking at these changes it looks like it's not there unless we're running in `aarch64`

Reply:
it's below since it's using absolute positioniong anyway so moved it one level